spiceaidocs/docs/api/auth/index.md

```diff
@@ -0,0 +1,63 @@
+---
+title: 'Authentication'
+sidebar_label: 'Authentication'
+sidebar_position: 3
+description: 'Authentication documentation'
+pagination_prev: null
+pagination_next: null
+---
+
+Spice supports adding optional authentication to its API endpoints via configurable API keys.
+
+Use the `auth` section as a child to `runtime` to provide the API keys. Multiple API keys can be specified, and any of the keys can be used to authenticate requests.
+
+```yaml
+runtime:
+  auth:
+    api-key:
+      enabled: true
+      keys:
+        - ${ secrets:api_key } # Use the secret replacement syntax to load the API key from a secret store
+        - 1234567890 # Or specify the API key directly
+```
+
+To learn more about secrets, see [Secret Stores](../../components/secret-stores/index.md).
+
+:::info
+
+The API key authentication is applied on startup and changes will not take effect until the runtime is restarted.
+
+:::
+
+## HTTP
+
+For HTTP routes, the API key is expected to be included in the `X-API-Key` header.
+
+```bash
```

Comment:
I though HTTP endpoints support `Authorization: Bearer ${ api_key }` too?

Reply:
No, they only look at the `X-API-Key` header. The Flight endpoint looks at the `Authorization` header to comply with the Flight BasicAuth pattern, but not the HTTP endpoints.